### DISCUSSION TITLE
Communication with Ultrasonix machine

### DISCUSSION THREAD

**Discussion by SS-Robotics** [PlusToolkit/PlusLib#988] [Q&A]
Hi,

I have recently started working with an old Ultrasonix machine. I want to use Plus and Ulterius SDK to collect RF data directly from the Exam software. I am going to use another computer for data collection. I have tried to connect the two systems(Ultrasonix machine and data collection computer) via Ethernet. Unfortunately, I couldn't run the provided release of Plus on another computer. So I tried to make sure that the two systems are connected. But even when I try to connect the two systems through Ulterius demo software I got connection errors. I am not sure if I am missing any steps in setting up the connection. I wonder if there is any documentation about how to set up the connection between two computers and then start collecting data. I couldn't find anything related to that in your website. I am using SonixTouch. The SDK and software versions are 6.1.1. I would be grateful if anyone could give some information about this.

Thanks,
S F

**Comment by lassoan** [comment]
> But even when I try to connect the two systems through Ulterius demo software I got connection errors

To simplify the problem as much as possible, I would recommend to start with trying to make the Ulterius demo application that comes with the 6.1.1 SDK, on the Ultrasonix PC.

Make sure the Exam software is running. You may also need to switch to research mode.

Make sure the streaming option is enabled (I don't remember the exact name of the option). To enable it, you need to have a valid license, purchased from Ultrasonix. Even if you had the license, it may have expired by now (we had to contact Ultrasonix occasionally to renew the license).

If you cannot even make the Ulterius demo work then you probably cannot use this system anymore. There are a number of other ultrasound systems that Plus supports - see http://perk-software.cs.queensu.ca/plus/doc/nightly/user/Configuration.html

**Reply by SS-Robotics** [reply]
Hi,

Regarding my previous question, I have tested connecting to Ulterius instead of PLUS on the Ultrasonix PC. One thing that I found out is that I can connect to the SDK and PLUS server if I downgrade my Exam software version to 5.7.5. However, if I want to use version 6.1.1, I will get the following error.
In my case, I cannot use the previous version of the Exam software on our Ultrasonix machine. I wonder whether you have tested PLUS with version 6.1.1 and whether you have encountered anything similar to this issue before. I would appreciate any suggestions.


![PLUS](https://user-images.githubusercontent.com/104461273/196297557-5dd879fa-ac01-441c-afbc-009bbd7f1de2.jpg)

**Comment by SS-Robotics** [comment]
Hi all,

I am still struggling with this. I wonder if anyone had the same experience before that can share to solve this issue.

Thanks,
S A
